### PR TITLE
docs(lsp): improve API documentation for inlay hint

### PR DESCRIPTION
**Pull request by ofseed**
The current documentation is slightly inaccurate or missing.

### CODE REVIEW

**Comment by MariaSolOs** [comment]
I don't think this makes the documentation clearer since it's not directly specified what "scope" here is referring to.

**Comment by ofseed** [comment]
@MariaSolOs Would this be better?

**Comment by MariaSolOs** [comment]
> it is mainly to fix bugs rather than provide more detailed descriptions.

? I thought that the purpose of this PR was to improve the documentation, but now you're saying it is a bugfix...?

**Comment by ofseed** [comment]
improve while fixing :) these are wrong

https://github.com/neovim/neovim/blob/8c7a8be27437d8f2d640daf1c2598d4c18933e04/runtime/lua/vim/lsp/inlay_hint.lua#L101-L101

https://github.com/neovim/neovim/blob/8c7a8be27437d8f2d640daf1c2598d4c18933e04/runtime/lua/vim/lsp/inlay_hint.lua#L405-L405

File: runtime/lua/vim/lsp/inlay_hint.lua [hunk 1]
```diff
@@ -402,14 +403,20 @@ end
 --- Buffer number, or 0 for current buffer, or nil for all.
 --- @field bufnr integer?
 
---- Enables or disables inlay hints for a buffer.
+--- Enables or disables inlay hints for the |filter|ed scope.
```
Comment (wookayin) [inline runtime/lua/vim/lsp/inlay_hint.lua:406]: What does `|filter|ed` scope precisely mean? The help entry `|filter|` doesn't seem relevant.

Comment (ofseed) [reply runtime/lua/vim/lsp/inlay_hint.lua]: Oh yes, it's a misuse exactly. Would the new way of writing this be better?

File: runtime/doc/lsp.txt [hunk 2]
```diff
@@ -1589,12 +1589,16 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
 Lua module: vim.lsp.inlay_hint                                *lsp-inlay_hint*
 
 enable({enable}, {filter})                       *vim.lsp.inlay_hint.enable()*
-    Enables or disables inlay hints for a buffer.
+    Enables or disables inlay hints for the {filter}ed scope.
 
-    To "toggle", pass the inverse of `is_enabled()`: >lua
+    To "toggle" globally, pass the inverse of `is_enabled()`: >lua
```
Comment (justinmk) [inline runtime/doc/lsp.txt:1594]: this sentence isn't about scope, it's about toggling. toggling is always done with is_enabled, regardless of scope.

the example happens to set global scope, but that's irrelevant.
```suggestion
    To "toggle", pass the inverse of `is_enabled()`: >lua
```

File: runtime/doc/lsp.txt [hunk 3]
```diff
@@ -1589,12 +1589,16 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
...
         vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled())
 <
 
+    Or you can disable it individually for the current buffer: >lua
+        vim.lsp.inlay_hint.enable(false, { bufnr = 0 })
```
Comment (justinmk) [inline runtime/doc/lsp.txt:1599]: this is over-explaining. examples show mechanics of usage. they generally don't need to demonstrate usage of each parameter, unless the parameter is unusual.

**Comment by dundargoc** [comment]
Thank you for your contribution! We collect all small documentation fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/28609

**Pull request closed without merging**